src/f32-rsum/gen/f32-rsum-rvv-u2v.c

```diff
@@ -56,20 +56,22 @@ void xnn_f32_rsum_ukernel__rvv_u2v(
       vfloat32m2_t in1_f32v = __riscv_vle32_v_f32m2(input, n); input += n;
       vfloat32m2_t in2_f32v = __riscv_vle32_v_f32m2(input, n); input += n;
       vfloat32m2_t in3_f32v = __riscv_vle32_v_f32m2(input, n); input += n;
-      vfloat32m2_t sum0_f32v = __riscv_vfadd_vv_f32m2(in0_f32v, in1_f32v, n);
-      vfloat32m2_t sum1_f32v = __riscv_vfadd_vv_f32m2(in2_f32v, in3_f32v, n);
-      sum_f32v = __riscv_vfadd_vv_f32m2(sum_f32v, sum0_f32v, n);
-      sum_f32v = __riscv_vfadd_vv_f32m2(sum_f32v, sum1_f32v, n);
+      vfloat32m2_t sum01_f32v = __riscv_vfadd_vv_f32m2(in0_f32v, in1_f32v, n);
+      vfloat32m2_t sum23_f32v = __riscv_vfadd_vv_f32m2(in2_f32v, in3_f32v, n);
+      sum0_f32v = __riscv_vfadd_vv_f32m2(sum0_f32v, sum01_f32v, n);
+      sum1_f32v = __riscv_vfadd_vv_f32m2(sum1_f32v, sum23_f32v, n);
     }
   }
+  vfloat32m2_t sum_f32v = __riscv_vfadd_vv_f32m2(sum0_f32v, sum1_f32v, n);
   if (batch > 0) {
-    for (; batch > 0; batch -= n) {
-      n = __riscv_vsetvl_e32m2(batch);
-      vfloat32m2_t in_f32v = __riscv_vle32_v_f32m2(input, n); input += n;
-      sum_f32v = __riscv_vfadd_vv_f32m2(sum_f32v, in_f32v, n);
+    for (; batch > 0;) {
+      size_t n1 = __riscv_vsetvl_e32m2(batch);
+      vfloat32m2_t in_f32v = __riscv_vle32_v_f32m2(input, n1); input += n1;
+      sum_f32v = __riscv_vfadd_vv_f32m2(sum_f32v, in_f32v, n1);
+      batch -= n1;
     }
   }
-  vfloat32m2_t out_f32v = __riscv_vfmul_vf_f32m2(sum_f32v, params->scalar.scale, n_sum);
-  acc_f32v = __riscv_vfredosum_vs_f32m2_f32m1(out_f32v, acc_f32v, n_sum);
+  vfloat32m2_t out_f32v = __riscv_vfmul_vf_f32m2(sum_f32v, params->scalar.scale, n);
+  acc_f32v = __riscv_vfredosum_vs_f32m2_f32m1(out_f32v, acc_f32v, n);
```

Comment:
nit: The rest of the algorithm doesn't care about order. This is not going to be the performance bottleneck, but it fits in more with the rest of the code if this is the unordered reduction `__riscv_vfredusum_vs_f32m2_f32m1`

Similarly for the other LMUL implementations